pkg/storage/chunk/azure/blob_storage_client.go

```diff
@@ -255,29 +254,87 @@ func (b *BlobStorage) newPipeline(hedgingCfg hedging.Config, hedging bool) (pipe
 			MaxRetryDelay: b.cfg.MaxRetryDelay,
 		},
 	}
-	client := defaultClientFactory()
 
-	opts.HTTPSender = pipeline.FactoryFunc(func(next pipeline.Policy, po *pipeline.PolicyOptions) pipeline.PolicyFunc {
-		return func(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
-			resp, err := client.Do(request.WithContext(ctx))
-			return pipeline.NewHTTPResponse(resp), err
-		}
-	})
-
-	if hedging {
-		client, err := hedgingCfg.ClientWithRegisterer(client, prometheus.WrapRegistererWithPrefix("loki", prometheus.DefaultRegisterer))
+	if !b.cfg.UseManagedIdentity {
+		credential, err := azblob.NewSharedKeyCredential(b.cfg.AccountName, b.cfg.AccountKey.Value)
 		if err != nil {
 			return nil, err
 		}
+
+		client := defaultClientFactory()
+
 		opts.HTTPSender = pipeline.FactoryFunc(func(next pipeline.Policy, po *pipeline.PolicyOptions) pipeline.PolicyFunc {
 			return func(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
 				resp, err := client.Do(request.WithContext(ctx))
 				return pipeline.NewHTTPResponse(resp), err
 			}
 		})
+
+		if hedging {
+			client, err := hedgingCfg.ClientWithRegisterer(client, prometheus.WrapRegistererWithPrefix("loki", prometheus.DefaultRegisterer))
+			if err != nil {
+				return nil, err
+			}
+			opts.HTTPSender = pipeline.FactoryFunc(func(next pipeline.Policy, po *pipeline.PolicyOptions) pipeline.PolicyFunc {
+				return func(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
+					resp, err := client.Do(request.WithContext(ctx))
+					return pipeline.NewHTTPResponse(resp), err
+				}
+			})
+		}
+		return azblob.NewPipeline(credential, opts), nil
```

Comment:
I think the only change required was this credentials object, all other options should be the same wether you used managed identitfy or not.

I would remove this return, use the same opts and credentials object, move the if down and override the default credentials when required.

Reply:
Thank you, do you want me to make the change & push it?

Reply:
Yeah I think it's just about moving the code around to ensure the client is configure for all type of credentials.

Reply:
OK, I pushed another commit, could you take a look if I am correct..? Thank you so much!!